apps/morphologyServer.cpp

```diff
@@ -0,0 +1,84 @@
+
+/*
+ * Copyright (c) 2017, EPFL/Blue Brain Project
+ *                     Stefan.Eilemann@epfl.ch
+ */
+
+#include <brion/constants.h>
+#include <brion/morphology.h>
+#include <brion/plugin/morphologyZeroEQ.h>
+#include <keyv/Map.h>
+#include <lunchbox/daemon.h>
+#include <lunchbox/log.h>
+#include <zeroeq/server.h>
+#include <zeroeq/uri.h>
+
+int main(const int argc, char* argv[])
+{
+    zeroeq::Server server(argc == 2 ? zeroeq::URI(argv[1]) : zeroeq::URI());
+    const std::string address = server.getURI().getHost() + ":" +
+                                std::to_string(int(server.getURI().getPort()));
+    std::cout
+        << "Morphology server reachable on:" << std::endl
+        << "  export ZEROEQ_SERVERS=\"" << address << ",$ZEROEQ_SERVERS\""
+        << std::endl
+        << "  zeroeq://" << address << "/path/to/morphology" << std::endl
+        << std::endl
+        << "  [c]ache read, [d]isk read with cache update, disk read "
+           "with cache [e]rror, [D]isk read uncached, morphology [l]oad error: "
+        << std::flush;
+    lunchbox::Log::setOutput(std::string(argv[0]) + ".log");
+
+    auto cache = keyv::Map::createCache();
+
+    server.handle(brion::ZEROEQ_GET_MORPHOLOGY, [&](const void* data,
+                                                    const size_t size) {
+        if (!data || !size)
+            return zeroeq::ReplyData();
+
+        const std::string path((const char*)data, size);
+        if (cache)
+        {
+            servus::Serializable::Data value;
+            cache->takeValues({path},
+                              [&](const std::string&, char* d, const size_t s) {
+                                  value.ptr.reset(d);
+                                  value.size = s;
+                              });
+
+            if (value.ptr && value.size)
+            {
+                std::cout << 'c' << std::flush;
+                return zeroeq::ReplyData(brion::ZEROEQ_GET_MORPHOLOGY, value);
```

Comment:
It returns the reply ID, not the request ID (which in this use case is identical to the request ID). A handler may return different reply IDs to the same request, e.g., for error codes or different return value types.